libs/books/data-access/src/lib/+state/reading-list.reducer.ts

```diff
@@ -58,7 +58,17 @@ const readingListReducer = createReducer(
   ),
   on(ReadingListActions.failedRemoveFromReadingList, (state, action) =>
     readingListAdapter.addOne({ ...action.item }, state)
-  )
+  ),
+  on(ReadingListActions.markBookAsFinished, (state, action) => {
+    const updatedItem: Update<ReadingListItem> = {
+      id: action.item.bookId,
+      changes: {
+        finished: true,
+        finishedDate: new Date().toISOString()
```

Comment:
Are you sure you should be setting this in a reducer? Ideally this should already be in the action when you receive it. Setting it here and in the effect is going to result in two values that are out of sync.

Reply:
Addressed it